### PR TITLE
Batch rust update

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
# Updated dependencies

- [x] Bump `diesel` from `2.0.3` to `2.1.0` <https://github.com/diesel-rs/diesel/compare/v2.0.3...v2.1.0>
- [x] Bump `digest` from `0.10.6` to `0.10.7` <https://github.com/RustCrypto/traits/compare/v0.10.6...v0.10.7>
- [x] Bump `futures-lite` from `1.12.0` to `1.13.0` <https://github.com/smol-rs/futures-lite/compare/v1.12.0...v1.13.0>
- [x] Bump `getrandom_v2` from `0.2.3` to `0.2.10` <https://github.com/rust-random/getrandom/compare/v0.2.3...v0.2.10>
- [x] Bump `glob` from `0.3.0` to `0.3.1` <https://github.com/rust-lang/glob/compare/v0.3.0...v0.3.1>
- [x] Bump `hyper` from `0.14.26` to `0.14.27` <https://github.com/hyperium/hyper/compare/v0.14.26...v0.14.27>
- [x] Bump `itertools` from `0.10.5` to `0.11.0` <https://github.com/rust-itertools/itertools/compare/v0.10.5...v0.11.0>
- [x] Bump `js-sys` from `0.3.59` to `0.3.64` <https://github.com/rustwasm/wasm-bindgen/compare/v0.3.59...v0.3.64>
- [x] Bump `libsqlite3-sys` from `0.25.2` to `0.26.0` <https://github.com/rusqlite/rusqlite/compare/v0.25.2...v0.26.0>
- [x] Bump `log` from `0.4.17` to `0.4.19` <https://github.com/rust-lang/log/compare/v0.4.17...v0.4.19>
- [x] Bump `miniserde` from `0.1.30` to `0.1.31` <https://github.com/dtolnay/miniserde/compare/v0.1.30...v0.1.31>
- [x] Bump `once_cell` from `0.17.1` to `1.18.0` <https://github.com/matklad/once_cell/compare/v0.17.1...v1.18.0>
- [x] Bump `paste` from `1.0.12` to `1.0.13` <https://github.com/dtolnay/paste/compare/v1.0.12...v1.0.13>
- [x] Bump `percent-encoding` from `2.2.0` to `2.3.0` <https://github.com/servo/rust-url/compare/v2.2.0...v2.3.0>
- [x] Bump `pretty_assertions` from `1.3.0` to `1.4.0` <https://github.com/rust-pretty-assertions/rust-pretty-assertions/compare/v1.3.0...v1.4.0>
- [x] Bump `proc-macro2` from `1.0.56` to `1.0.64` <https://github.com/dtolnay/proc-macro2/compare/v1.0.56...v1.0.64>
- [x] Bump `pyo3` from `0.17.3` to `0.19.1` <https://github.com/pyo3/pyo3/compare/v0.17.3...v0.19.1>
- [x] Bump `quote` from `1.0.28` to `1.0.29` <https://github.com/dtolnay/quote/compare/v1.0.28...v1.0.29>
- [x] Bump `regex` from `1.8.4` to `1.9.1` <https://github.com/rust-lang/regex/compare/v1.8.4...v1.9.1>
- [x] Bump `reqwest` from `0.11.17` to `0.11.18` <https://github.com/seanmonstar/reqwest/compare/v0.11.17...v0.11.18>
- [x] Bump `rsa` from `0.7.2` to `0.8.2` <https://github.com/RustCrypto/RSA/compare/v0.7.2...v0.8.2>
- [x] Bump `rstest_reuse` from `0.4.0` to `0.6.0` <https://github.com/la10736/rstest_reuse/compare/v0.4.0...v0.6.0>
- [x] Bump `rstest` from `0.17.0` to `0.18.1` <https://github.com/la10736/rstest/compare/v0.17.0...v0.18.1>
- [x] Bump `serde_bytes` from `0.11.9` to `0.11.11` <https://github.com/serde-rs/bytes/compare/v0.11.9...v0.11.11>
- [x] Bump `serde_json` from `1.0.96` to `1.0.100` <https://github.com/serde-rs/json/compare/v1.0.96...v1.0.100>
- [x] Bump `serde_test` from `1.0.159` to `1.0.171` <https://github.com/serde-rs/serde/compare/v1.0.159...v1.0.171>
- [x] Bump `serde` from `1.0.164` to `1.0.171` <https://github.com/serde-rs/serde/compare/v1.0.164...v1.0.171>
- [x] Bump `serde_with` from `2.3.2` to `3.0.0` <https://github.com/jonasbb/serde_with/compare/v2.3.2...v3.0.0>
- [x] Bump `sha2` from `0.10.6` to `0.10.7` <https://github.com/RustCrypto/hashes/compare/v0.10.6...v0.10.7>
- [x] Bump `syn` from `2.0.13` to `2.0.25` <https://github.com/dtolnay/syn/compare/v2.0.13...v2.0.25>
- [x] Bump `thiserror` from `1.0.40` to `1.0.43` <https://github.com/dtolnay/thiserror/compare/v1.0.40...v1.0.43>
- [x] Bump `tokio` from `1.28.1` to `1.29.1` <https://github.com/tokio-rs/tokio/compare/v1.28.1...v1.29.1>
- [x] Bump `url` from `2.3.1` to `2.4.0` <https://github.com/servo/rust-url/compare/v2.3.1...v2.4.0>
- [x] Bump `uuid` from `1.2.2` to `1.4.0` <https://github.com/uuid-rs/uuid/compare/v1.2.2...v1.4.0>
- [x] Bump `wasm-bindgen-futures` from `0.4.33` to `0.4.37` <https://github.com/rustwasm/wasm-bindgen/compare/v0.4.33...v0.4.37>
- [x] Bump `wasm-bindgen-test` from `0.3.33` to `0.3.37` <https://github.com/rustwasm/wasm-bindgen/compare/v0.3.33...v0.3.37>
- [x] Bump `wasm-bindgen` from `0.2.86` to `0.2.87` <https://github.com/rustwasm/wasm-bindgen/compare/v0.2.86...v0.2.87>
- [x] Bump `web-sys` from `0.3.63` to `0.3.64` <https://github.com/rustwasm/wasm-bindgen/compare/v0.3.63...v0.3.64>
- [x] Bump `x25519-dalek` from `2.0.0-rc.2` to `2.0.0-rc.3` <https://github.com/dalek-cryptography/x25519-dalek/compare/v2.0.0-rc.2...v2.0.0-rc.3>

### CODE REVIEW

File: libparsec/crates/crypto/src/rustcrypto/sequester.rs [hunk 1]
```diff
@@ -317,11 +317,13 @@ impl SequesterVerifyKeyDer {
         let (signature, data) =
             deserialize_with_armor(data, self.size_in_bytes(), Self::ALGORITHM)?;
 
-        // TODO: It Seems to be a mistake from RustCrypto/RSA
-        // Why should we allocate there ?
-        self.0
-            .verify(data, &Signature::from(signature.to_vec()))
-            .map_err(|_| CryptoError::SignatureVerification)?;
+        Signature::try_from(signature)
+            .map_err(|_| CryptoError::Signature)
```
Comment (vxgmichel) [inline libparsec/crates/crypto/src/rustcrypto/sequester.rs:321]: This is  a new kind  of error that can occur, do we have a corresponding test ? how is it handled in the sodiumoxide implementation ?

**Review by vxgmichel — CHANGES_REQUESTED** [review]

File: libparsec/crates/crypto/src/rustcrypto/sequester.rs (hunk 1 above)
Comment (FirelightFlagboy) [reply libparsec/crates/crypto/src/rustcrypto/sequester.rs:321]: `Signature::try_from` won't raise an error from the [code implementation](https://docs.rs/rsa/0.8.2/src/rsa/pss.rs.html#135-143) so I wouldn't be able to provide a test that fail to parse a raw signature bytes

Comment (FirelightFlagboy) [reply libparsec/crates/crypto/src/rustcrypto/sequester.rs:321]: https://github.com/RustCrypto/RSA/issues/349

Comment (vxgmichel) [reply libparsec/crates/crypto/src/rustcrypto/sequester.rs:321]: a comment should be added to explain this then

**Review by vxgmichel — APPROVED** [review]
Approved by @touilleMan

**Review by mmmarcos — APPROVED** [review]
LGTM

**Pull request merged**